Neos.Utility.Files/Classes/Files.php

```diff
@@ -19,6 +19,34 @@
  */
 abstract class Files
 {
+    /**
+     * Duckcheck if a path looks like it is absolute.
+     */
+    public static function isAbsolutePath(string $path): bool
+    {
+        if ($path === '') {
+            return false;
+        }
+        $isAbsolutePathInWindowsStyle = preg_match('`^[a-zA-Z]:[/\\\\][^/\\\\]`', $path) === 1;
+        $isAbsolutePathInUnixStyle = str_starts_with($path, '/');
+        return $isAbsolutePathInUnixStyle || $isAbsolutePathInWindowsStyle;
+    }
+
+    /**
+     * Duckcheck if a path looks like it is fully qualified.
+     * That means its absolute and has no directory traversal.
+     * {@see realpath()}
+     */
+    public static function isRealPath(string $path): bool
```

Comment:
This could be misleading as the linux `realpath` has another important function, which is to resolve symlinks, and we do not know if this path contains symlinks. As in `/var/www/app/current/Web` vs `/var/www/app/20231219/Web`

Reply:
yes maybe `isAbsoluteUnixStyleFilePathWithoutDirectoryTraversal` ?

This might still be useful just to be able to do a simple optional assertion, so we notice more quickly if something is wrong.

https://github.com/neos/neos-development-collection/pull/4838

Reply:
or `isSimpleAbsoluteUnixPath` `isPureUnixPath` or `isCleanUnixPath`? idk

Reply:
Yes and the symlinking thing is exactly what lead to this [bug](https://github.com/neos/neos-development-collection/issues/4915) rendering the helper obsolete.

So I guess we only need the first helper